superset/utils.py

```diff
@@ -252,49 +267,76 @@ def init(superset):
         'database_access',
         'muldelete',
     ])
+    READ_ONLY_PRODUCT = set(
+        product(READ_ONLY_PERMISSION, READ_ONLY_MODELVIEWS))
 
     db = superset.db
     models = superset.models
     config = superset.app.config
     sm = superset.appbuilder.sm
+
+    # Creating default roles
     alpha = sm.add_role("Alpha")
     admin = sm.add_role("Admin")
+    gamma = sm.add_role("Gamma")
+    public = sm.add_role("Public")
+    sql_lab = sm.add_role("sql_lab")
+
     get_or_create_main_db(superset)
 
+    # Global perms
     merge_perm(sm, 'all_datasource_access', 'all_datasource_access')
+    merge_perm(sm, 'all_database_access', 'all_database_access')
 
     perms = db.session.query(ab_models.PermissionView).all()
-    # set alpha and admin permissions
-    for perm in perms:
-        if (
-                perm.permission and
-                perm.permission.name in ('datasource_access', 'database_access')):
-            continue
-        if (
-                perm.view_menu and
-                perm.view_menu.name not in ADMIN_ONLY_VIEW_MENUES and
-                perm.permission and
-                perm.permission.name not in ADMIN_ONLY_PERMISSIONS):
+    perms = [p for p in perms if p.permission and p.view_menu]
 
-            sm.add_permission_role(alpha, perm)
-        sm.add_permission_role(admin, perm)
+    # set admin perms
+    for p in perms:
+        sm.add_permission_role(admin, p)
 
-    gamma = sm.add_role("Gamma")
-    public_role = sm.find_role("Public")
-    public_role_like_gamma = \
-        public_role and config.get('PUBLIC_ROLE_LIKE_GAMMA', False)
+    # set alpha perms
+    for p in perms:
+        if (
+                (
+                    p.view_menu.name not in ADMIN_ONLY_VIEW_MENUES and
+                    p.permission.name not in ADMIN_ONLY_PERMISSIONS
+                ) or
+                (p.permission.name, p.view_menu.name) in READ_ONLY_PRODUCT
+            ):
+            sm.add_permission_role(alpha, p)
+        else:
+            sm.del_permission_role(alpha, p)
+
+    # set gamma permissions and public to be alike if specified
+    PUBLIC_ROLE_LIKE_GAMMA = config.get('PUBLIC_ROLE_LIKE_GAMMA', False)
+    for p in perms:
+        if (
+                (
+                    p.view_menu.name not in ADMIN_ONLY_VIEW_MENUES and
+                    p.permission.name not in ADMIN_ONLY_PERMISSIONS and
+                    p.permission.name not in ALPHA_ONLY_PERMISSIONS
+                ) or
+                (p.permission.name, p.view_menu.name) in READ_ONLY_PRODUCT
+        ):
+            sm.add_permission_role(gamma, p)
+            if PUBLIC_ROLE_LIKE_GAMMA:
+                sm.add_permission_role(public, p)
+        else:
```

Comment:
I'm thinking we move to fully control these base roles, if people alter them they will be overwritten. People should create and compose new roles in their environment instead of altering the ones provided by Superset